cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_utils_test.go

```diff
@@ -292,6 +294,181 @@ valueWithDefault: 80
+					Stderr: "inspectStderr",
+				},
+			},
+		}, `#### Deploy
+
+<x60><x60><x60>
+deployStdout
+<x60><x60><x60>
+
+#### Fetch
+
+<x60><x60><x60>
+fetchStdout
+<x60><x60><x60>
+
+### Errors
+
+#### Deploy
+
+<x60><x60><x60>
+deployStderr
+<x60><x60><x60>
+
+#### Fetch
+
+<x60><x60><x60>
+fetchStderr
+<x60><x60><x60>
+
+`},
+		{"renders the expected notes (no stderr)", &kappctrlv1alpha1.App{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       appResource,
+				APIVersion: kappctrlAPIVersion,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "my-installation",
+			},
+			Spec: kappctrlv1alpha1.AppSpec{
+				SyncPeriod: &metav1.Duration{(time.Second * 30)},
+			},
+			Status: kappctrlv1alpha1.AppStatus{
+				Deploy: &kappctrlv1alpha1.AppStatusDeploy{
+					Stdout: "deployStdout",
+				},
+				Fetch: &kappctrlv1alpha1.AppStatusFetch{
+					Stdout: "fetchStdout",
+				},
+				Inspect: &kappctrlv1alpha1.AppStatusInspect{
+					Stdout: "inspectStdout",
+				},
+			},
+		}, `#### Deploy
+
+<x60><x60><x60>
+deployStdout
+<x60><x60><x60>
+
+#### Fetch
+
+<x60><x60><x60>
+fetchStdout
+<x60><x60><x60>
+
+`},
+		{"renders the expected notes (no stdout)", &kappctrlv1alpha1.App{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       appResource,
+				APIVersion: kappctrlAPIVersion,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "my-installation",
+			},
+			Spec: kappctrlv1alpha1.AppSpec{
+				SyncPeriod: &metav1.Duration{(time.Second * 30)},
+			},
+			Status: kappctrlv1alpha1.AppStatus{
+				Deploy: &kappctrlv1alpha1.AppStatusDeploy{
+					Stderr: "deployStderr",
+				},
+				Fetch: &kappctrlv1alpha1.AppStatusFetch{
+					Stderr: "fetchStderr",
+				},
+				Inspect: &kappctrlv1alpha1.AppStatusInspect{
+					Stderr: "inspectStderr",
+				},
+			},
+		}, `### Errors
+
+#### Deploy
+
+<x60><x60><x60>
+deployStderr
+<x60><x60><x60>
+
+#### Fetch
+
+<x60><x60><x60>
+fetchStderr
+<x60><x60><x60>
+
+`},
+		{"renders the expected notes (missing field)", &kappctrlv1alpha1.App{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       appResource,
+				APIVersion: kappctrlAPIVersion,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "default",
+				Name:      "my-installation",
+			},
+			Spec: kappctrlv1alpha1.AppSpec{
+				SyncPeriod: &metav1.Duration{(time.Second * 30)},
+			},
+			Status: kappctrlv1alpha1.AppStatus{
+				Fetch: &kappctrlv1alpha1.AppStatusFetch{
+					Stdout: "fetchStdout",
+				},
+				Inspect: &kappctrlv1alpha1.AppStatusInspect{
+					Stdout: "inspectStdout",
+				},
+			},
+		}, `#### Fetch
+
+<x60><x60><x60>
+fetchStdout
+<x60><x60><x60>
+
+`},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			readme := buildPostInstallationNotes(tt.app)
+			// when using `` we cannot escape the ` char itself, so let's replace it later
+			expected := strings.ReplaceAll(tt.expected, "<x60>", "`")
+			if want, got := expected, readme; !cmp.Equal(want, got) {
+				t.Logf(got)
```

Comment:
Debugging output?

Reply:
Ooops :innocent:, forgot to remove it. Thanks!